src/main/resources/alma/fix/describedBy.fix

```diff
@@ -106,6 +106,8 @@ do once("provenanceLinksMacro")
       replace_all("describedBy.resultOf.object.sourceOrganization.id", "(.*)\\/NWBib$", "DE-$1")
     elsif any_match("$[field]",".*(dnb|DNB|GWDNB|GWDEB|GWDDB|DEDNM|DEDND).*")
       replace_all("$[field]", "^.*$", "DE-101")
+    elsif any_match("$[field]","^38\\/\\d+$")
+      replace_all("$[field]", "^38\\/(\\d+)$", "38-$1")
```

Comment:
I thought about this approach, and one question that came into my mind is: Can we be sure that slashes are always synonymous to dashes? Are there some cases (some libraries?) that use slashes to represent underscores? Asterisks?

Reply:
As you noted : slashes just breaks any URL. (BTW I don't know why anyone on earth would want to catalogue that data _not_ as an ISIL (and see here the regex for a valid ISIL https://www.wikidata.org/wiki/Property:P791)).
Anyway: if a library might run into problems with that healing of their data they can open an issue here for further healing.

Reply:
The regular expression you quoted explicitly *allows* slashes (but the Wikidata article doesn't quote any references)
![image](https://github.com/hbz/lobid-resources/assets/6136469/c3f1e108-adda-40c0-8685-73e37e83f37a)

Reply:
Nonetheless, the remark about URLs is a good point. I have implemented a more general "slash to dash" approach now, as per https://github.com/metafacture/metafacture-fix#str_contain

Reply:
haha - eh, right :blush: ooohhkay - seems to be best to URL-Encode/ Percent Escape the slashes, as they are allowed, as you pointed out, and don't just substitute them with a dash.
And as `DE-38-459` seems to be the correct ISIL (see https://sigel.staatsbibliothek-berlin.de/aenderungen-mitteilen/?isil=DE-38-459 ) it seems to me that the underlying AMA data should be fixed.

Reply:
URLEncode is a great point. Updated. :smile:

Reply:
But now, 38/459 from the ALMA data is being converted to `38%2F459`. Which is still an incorrect ISIL. This is not problematic because our application doesn't immediately rely on this URL, but it's still inconsistent.

Reply:
Right. Therefore, I think there is only one reasonable fix: your institution has to fix the data they catalog to use the proper ISIL of their institution. Can/will they do that?

Reply:
That depends. Is MARC `040` _supposed_ to contain an ISIL in the first place? If the MARC standard doesn't mandate the correct ISIL, our institution will probably argue that `38/459` is a perfectly valid "internal identifier" and that it's "our problem" if we try to interpret this data as ISIL.

cf. http://www.alma-dach.org/alma-marc/bibliographic/040/040.html which states `MARC-Code oder Name der Organisation`

Reply:
wait wait wait. I think this just refers to the Sigil and not the proper ISIL, because of that it has the  slash instead of a dash.

Reply:
https://github.com/hbz/lobid-resources/blob/714-rpb/src/main/resources/sigel2isilMap.csv

Reply:
> I think this just refers to the Sigil and not the proper ISIL, because of that it has the slash instead of a dash.

Which comes back around to my argument: It is "the problem/fault of Lobid" for trying to interpret `040a` as ISIL (which can subsequently be used in the URL for lobid-resources) when in reality the source data was never supposed to be an ISIL.

Reply:
https://github.com/hbz/lobid-resources/blob/ffbae9b984af3163ccead44edae285f287717a84/src/main/resources/sigel2isilMap.csv#L1546

Reply:
Okay, so now we're back to "slash to dash". What about your
> wait wait wait.

interjection? The `sigel2IsilMap` that you linked seems like an elegant solution, but it is no longer in the `master` git tree...

Reply:
i didnt want you to go on discussing ISILs :D
The map is not viable. But perhaps we could use it. Let me check.

Reply:
I will open an PR to replace this with a lookup based on the ISILs mapping.